lib/internal/http2/core.js

```diff
@@ -3380,8 +3380,10 @@ function connect(authority, options, listener) {
 ObjectDefineProperty(connect, promisify.custom, {
   __proto__: null,
   value: (authority, options) => {
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       const server = connect(authority, options, () => resolve(server));
+
+      server.once('error', reject);
```

Comment:
It would be good to remove this error listener immediately before calling `resolve()`.

It's useful to do so because otherwise this will silently swallow any thrown connection errors (calling reject, which will do nothing once this is resolved) after the initial connection.